Number tables

bwhelm opened the pull request:
As the first commit indicates, this allows for table captions with attributes. It seems to work reasonably well for LaTeX, html, and docx.

Notice that `latex_table()` is a big hack. I'd appreciate better ideas for how to insert an appropriate `\label{id}` command into the LaTeX generated by pandoc.

bwhelm committed:
Allow for table captions with attributes

Tables preceded or followed by captions like this:

```
: Table Caption {#label}
```

will get appropriate numbering and cross-references in latex, html,
docx, and markdown.

bwhelm committed:
Not needed: table captions already retain formatting.

bwhelm committed:
Miscellaneous clean-up:

1. Alter metadata if table or figure exists

2. Allow for unnumbered table caption

aaren commented:
Table referencing was something that I meant to add, but didn't get round to.

This solution is a great hack and makes sense given the current limitations of the code. However, it might be possible to do some higher level restructuring to make this a bit more maintainable.

Also, would be good to keep compatibility with [pandoc-crossref tables](https://github.com/lierdakil/pandoc-crossref#table-labels) in terms of syntax.

bwhelm committed:
Fix stupid mistake.

Not sure why this was in here, but it was deleting final 'u's from
every table cell. This fixes it.
